proposed/2021/BulkRestoreCoordination.md

```diff
@@ -0,0 +1,165 @@
+# Bulk restore coordination
+
+- [Nikolche Kolev](https://github.com/nkolev92)
+- Start Date 2021-05-03
+- GitHub Issue [10678](https://github.com/NuGet/Home/issues/10678)
+- GitHub PR [4058](https://github.com/NuGet/NuGet.Client/pull/4058)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+When a solution with SDK based projects is loaded, NuGet uses a sliding window to coordinate the number of restores being run to avoid extra restore/design time build loops.

```

Comment:
Thank you creating this spec.
Could be able to include graph for possible nomination paths could happen for both before and after this change?
We don't have full context for this?

Reply:
> Could be able to include graph for possible nomination paths could happen for both before and after this change?

I'm not sure I follow.

Currently the implementation is a sliding window. 
Effectively, NuGet waits for all projects to be nominated or no nomination for 10s before it starts a restore.

Reply:
Here you mentioned design time build. What are the other builds? When does other builds happen in which order?
Simple graph would be enough?

Reply:
> Here you mentioned design time build. What are the other builds? 

I [gave a presentation](https://msit.microsoftstream.com/embed/channel/8855a1ff-0400-a936-24ca-f1eaa053d9d5?app=microsoftteams&sort=undefined&l=en-us#) on that topic, please take look. The thumbnail suggests you were a part of it :) 

The design time builds are what the project-system does that leads to a nomination. 

I added a few references add the bottom of the doc.

Reply:
Link refers all videos. What was the name of video? 
In general having simple diagram helpful for other people too, they don't have to read everything.

Reply:
The name is NuGet and intellisense stage in Visual Studio, fixed the [link](https://msit.microsoftstream.com/video/e449a1ff-0400-9887-1aca-f1eb55f40c4b?channelId=8855a1ff-0400-a936-24ca-f1eaa053d9d5). 

> In general having simple diagram helpful for other people too, they don't have to read everything.

https://github.com/nkolev92/restore-to-intellisense/ is closest thing to a write-up, and I added it as a reference.
It is a very broad topic, which is why I have to refer you to the video again. 

This is design that's been discussed with various SMEs on the perf, CPS and project-system teams.

Reply:
Ok. Pardon me for my short memory. 😭 
I believe your referenced links are enough.